flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -297,9 +318,279 @@ private static Object handleLogicalTypeSchema(Schema fieldSchema, Object value)
          * @param logicalTypeString String containing the name for Logical Schema Type.
          * @return Encoder Function which converts AvroSchemaField to {@link DynamicMessage}
          */
-        private static UnaryOperator<Object> getLogicalEncoder(String logicalTypeString) {
+        private static UnaryOperator<Object> getLogicalEncoder(
+                Schema fieldSchema, String logicalTypeString) {
             Map<String, UnaryOperator<Object>> mapping = new HashMap<>();
+            mapping.put(LogicalTypes.date().getName(), AvroSchemaHandler::convertDate);
+            mapping.put(
+                    LogicalTypes.decimal(1).getName(),
+                    value -> AvroSchemaHandler.convertBigDecimal(value, fieldSchema));
+            mapping.put(
+                    LogicalTypes.timestampMicros().getName(),
+                    value -> convertTimestamp(value, true, "Timestamp(micros/millis)"));
+            mapping.put(
+                    LogicalTypes.timestampMillis().getName(),
+                    value -> convertTimestamp(value, false, "Timestamp(micros/millis)"));
+            mapping.put(LogicalTypes.uuid().getName(), AvroSchemaHandler::convertUUID);
+            mapping.put(LogicalTypes.timeMillis().getName(), value -> convertTime(value, false));
+            mapping.put(LogicalTypes.timeMicros().getName(), value -> convertTime(value, true));
+            mapping.put(
+                    LogicalTypes.localTimestampMillis().getName(),
+                    value -> convertDateTime(value, false));
+            mapping.put(
+                    LogicalTypes.localTimestampMicros().getName(),
+                    value -> convertDateTime(value, true));
+            mapping.put("geography_wkt", AvroSchemaHandler::convertGeography);
+            mapping.put("Json", AvroSchemaHandler::convertJson);
             return mapping.get(logicalTypeString);
         }
+
+        // ---- Utilities to enable Conversions for Logical Types ------------
+        @VisibleForTesting
+        static String convertUUID(Object value) {
+            if (value instanceof UUID) {
+                return ((UUID) value).toString();
+            } else {
+                Preconditions.checkArgument(
+                        value instanceof String, "Expecting a value as String type.");
+                UUID.fromString((String) value);
+                return (String) value;
+            }
```

Comment:
Lets maintain consistency with usage of if-else of preconditions. We can continue if-else here.
```
            if (value instanceof UUID) {
                return ((UUID) value).toString();
            } else if (value instanceof String) {
                // Validating correctness of UUID received as string.
                UUID.fromString((String) value);
            } else {
                throw new IllegalArgumentException(...);
            }
```

Reply:
ACK, Fixed.